conjure-java-core/src/main/java/com/palantir/conjure/java/Options.java

```diff
@@ -188,36 +188,44 @@ default boolean preferObjectBuilders() {
     }
 
     /**
-     * Enabling this option will enable associating errors with endpoints, which currently causes parameters of these
-     * errors to be serialized as JSON. This is a wire break and can cause errors for clients that parse error
-     * parameters. Do not enable this.
+     * Warning: This is an experimental feature that may cause compilation failures!
      * <p>
-     * This flag has been temporarily added to help us iterate on a safer rollout strategy for endpoint-associated
-     * errors.
-     * <p>
-     * This flag will eventually be removed, and you should not use it without letting the Conjure maintainers know.
+     * If enabled, endpoints that have associated errors will return a result type: a sealed interface permitting
+     * subclasses for the endpoint's return value, and each endpoint error. Each endpoint error is a subclass of
+     * {@link com.palantir.dialogue.EndpointError}.
+     *<p>
+     * Producing JARs with this feature enabled will result in a compile-time break when consumers bump their dependency
+     * on the JAR. This is because the return types of every endpoint with associated errors will change.
      */
     @Value.Default
-    default boolean dangerousDoNotUseEnableEndpointAssociatedErrors() {
+    default boolean generateDialogueEndpointErrorResultTypes() {
         return false;
     }
 
     /**
-     * Enabling this flag will lead to the generation of Dialogue client interfaces that are able to send a header to
-     * servers specifying the Conjure error parameter serialization format they expect, and create custom exceptions for
-     * each error defined in the service's package.
+     * Makes immutable copies of collections for union and alias types, respecting the --nonNullCollections flag.
      */
     @Value.Default
-    @Beta
-    default boolean generateErrorParameterFormatRespectingDialogueInterfaces() {
+    default boolean defensiveCollections() {
         return false;
     }
 
     /**
-     * Makes immutable copies of collections for union and alias types, respecting the --nonNullCollections flag.
+     * Uses sealed classes for union type generation.
+     * <p>
+     * Warning: Disabling this flag after publishing union objects with this flag enabled, is an ABI break (additional
+     * methods which are generated for sealed unions would be removed).
      */
     @Value.Default
-    default boolean defensiveCollections() {
+    default boolean sealedUnions() {
+        return false;
+    }
+
+    /**
+     * If {@link #sealedUnions} is enabled, this enables visitor generation for back-compatibility.
+     */
+    @Value.Default
+    default boolean sealedUnionVisitors() {
         return false;
```

Comment:
Thoughts on having this be `true` by default? This way devs don't have to set two flags to enable this feature. When the visitors are removed, it will be really clear because there will be a flag in the Gradle file setting this flag to false.